tests/plugins/test_resolve_composes.py

```diff
@@ -242,6 +250,143 @@ def test_request_compose_for_modules(self, workflow, reactor_config_map):  # noq
+            (flexmock(ODCSClient)
+                .should_receive('start_compose')
+                .with_args(source_type='pulp', source=' '.join(pulp_repos),
+                           arches=[arch], sigkeys=[])
+                .and_return(pulp_composes[arch]).once)
+            (flexmock(ODCSClient)
+                .should_receive('wait_for_compose')
+                .with_args(pulp_id)
+                .and_return(pulp_composes[arch]).once)
+
+        mock_content_sets_config(workflow._tmpdir, content_set)
+
+        repo_config = dedent("""\
+            compose:
+                pulp_repos: true
+                packages:
+                - spam
+                - bacon
+                - eggs
+            """)
+        mock_repo_config(workflow._tmpdir, repo_config)
+        workflow.buildstep_plugins_conf[0]['args']['platforms'] = arches
+        if arches is not None:
+            tag_compose = deepcopy(ODCS_COMPOSE)
+            tag_compose['arches'] = ' '.join(arches)
+            (flexmock(ODCSClient)
+                .should_receive('start_compose')
+                .with_args(source_type='tag', source=KOJI_TAG_NAME, arches=arches,
+                           packages=['spam', 'bacon', 'eggs'], sigkeys=['R123'])
+                .and_return(tag_compose).once)
+            (flexmock(ODCSClient)
+                .should_receive('wait_for_compose')
+                .with_args(ODCS_COMPOSE_ID)
+                .and_return(tag_compose).once)
+        else:
+            (flexmock(ODCSClient)
+                .should_receive('start_compose')
+                .with_args(source_type='tag', source=KOJI_TAG_NAME,
+                           packages=['spam', 'bacon', 'eggs'], sigkeys=['R123'])
+                .and_return(ODCS_COMPOSE).once)
+            (flexmock(ODCSClient)
+                .should_receive('wait_for_compose')
+                .with_args(ODCS_COMPOSE_ID)
+                .and_return(ODCS_COMPOSE).once)
+
+        self.run_plugin_with_args(workflow, reactor_config_map=reactor_config_map,
+                                  pulp_platforms=pulp_arches, arches=arches)
+
+    def test_request_compose_for_pulp_no_container(self, workflow, reactor_config_map):  # noqa
+        pulp_compose = {
+            'id': 85,
+            'result_repo': ODCS_COMPOSE_REPO,
+            'result_repofile': ODCS_COMPOSE_REPO + '/pulp_compose-x86_64',
+            'source': KOJI_TAG_NAME,
+            'source_type': 'pulp',
+            'sigkeys': '',
+            'state_name': 'done',
+            'arches': 'x86_64',
+            'time_to_expire': ODCS_COMPOSE_TIME_TO_EXPIRE.strftime(ODCS_DATETIME_FORMAT),
+        }
+        (flexmock(ODCSClient)
+            .should_receive('start_compose')
+            .with_args(
+                source_type='pulp',
+                source='pulp is no good here',
+                arches=['x86_64'],
+                sigkeys=[])
+            .and_return(pulp_compose).never)
```

Comment:
If we don't ever want this to be called, we don't care what our mocked version should return. So we can remove `.and_return(pulp_compose)` and `pulp_compose = ...` above.

Reply:
Also, you probably want `.never()` instead of `.never` to actually set the expectation. Otherwise, you're just retrieving the function reference without invoking it.

Reply:
Fixed.